pass-import fails to build in nixos-unstable

**Describe the bug**

`nixos-rebuild build` fails due to

```
builder for '/nix/store/da2g2vg3fa1vnw08dfdw9i62y1szvj5s-pass-import-3.1.drv' failed with exit code 1
```

**To Reproduce**
Steps to reproduce the behavior:
1. Use nixos-unstable, or nixos-unstable-small
2. Build a derivation depending on `pass-import`, for example 
 * `pass.withExtensions (ext: with ext; [ pass-import ])`
 * or `nix-build -E 'with import <nixpkgs> { }; pkgs.callPackage ./pkgs/tools/security/pass/extensions/import.nix {}'`

**Expected behavior**
Build succeeds

**Screenshots**
```
=================================== FAILURES ===================================
_________________ TestExportKeepassInsert.test_keepass_binary __________________

self = <tests.exports.test_keepass.TestExportKeepassInsert testMethod=test_keepass_binary>
pw = <MagicMock name='getpass' id='140737292069808'>

    @patch("getpass.getpass")
    def test_keepass_binary(self, pw):
        """Testing: keepass insert & show binary file."""
        pw.return_value = self.masterpassword
        self._init_keepass()
        with open(os.path.join(tests.assets, 'pass.png'), 'rb') as file:
            data = file.read()
        entry = {'data': data, 'path': 'pass.png'}
        ref = copy.deepcopy(entry)

        with Keepass(self.prefix) as keepass:
            keepass.insert(entry)
            keepass.parse()
>       self.assertEqual(keepass.data[0]['data'], ref['data'])
E       IndexError: list index out of range

tests/exports/test_keepass.py:99: IndexError
_________________ TestExportKeepassInsert.test_keepass_insert __________________

self = <tests.exports.test_keepass.TestExportKeepassInsert testMethod=test_keepass_insert>
pw = <MagicMock name='getpass' id='140737292245744'>

    @patch("getpass.getpass")
    def test_keepass_insert(self, pw):
        """Testing: keepass insert."""
        pw.return_value = self.masterpassword
        self._init_keepass()
        entry = {
            'password': 'UuQHzvv6IHRIJGjwKru7',
            'login': 'lnqYm3ZWtm',
            'url': 'https://twitter.com',
            'website': 'https://pujol.io',
            'uuid': '44jle5q3fdvrprmaahozexy2pi',
            'otpauth': 'otpauth://totp/totp-secret?secret=JBSWY3DPEHPK3PXP&'
                       'issuer=alice@google.com&algorithm=SHA1&digits=6&per'
                       'iod=30',
            'path': 'Test/test'
        }
        ref = copy.deepcopy(entry)

        with Keepass(self.prefix) as keepass:
            keepass.insert(entry)
            keepass.parse()
>           data = keepass.data[0]
E           IndexError: list index out of range

tests/exports/test_keepass.py:66: IndexError
______________ TestExportKeepassInsert.test_keepass_insert_empty _______________

self = <tests.exports.test_keepass.TestExportKeepassInsert testMethod=test_keepass_insert_empty>
pw = <MagicMock name='getpass' id='140737292307184'>

    @patch("getpass.getpass")
    def test_keepass_insert_empty(self, pw):
        """Testing: keepass insert empty."""
        pw.return_value = self.masterpassword
        self._init_keepass()
        entry = {'path': 'test'}
        ref = copy.deepcopy(entry)

        with Keepass(self.prefix) as keepass:
            keepass.insert(entry)
            keepass.parse()
>           data = keepass.data[0]
E           IndexError: list index out of range

tests/exports/test_keepass.py:81: IndexError
________________________ TestParse.test_import_keepass _________________________

self = <tests.imports.test_parse.TestParse testMethod=test_import_keepass>
pw = <MagicMock name='getpass' id='140737287742560'>

    @patch("getpass.getpass")
    def test_import_keepass(self, pw):
        """Testing: parse method for Keepass Kdbx."""
        prefix = os.path.join(tests.db, 'keepass.kdbx')
        reference = tests.reference()
        pw.return_value = self.masterpassword
        with tests.cls('Keepass', prefix) as importer:
            importer.parse()
            data = importer.data

        # Manual cleaning
        reference.extend(REFERENCE_KDBX)
        for entry in data:
            if entry['title'] == 'news.ycombinator.com':
                entry['title'] = 'https://news.ycombinator.com'

>       self.assertImport(data, reference)

tests/imports/test_parse.py:125:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/__init__.py:187: in assertImport
    self.assertEqual(len(data), len(refdata))
E   AssertionError: 0 != 18
=============================== warnings summary ===============================
tests/main/test_pass.py: 15 warnings
tests/main/test_sanitychecks.py: 1 warning
  /nix/store/kb5bwm424v7rmm09c3wd2wjsx4zl1cwf-python3.8-python-magic-0.4.22/lib/python3.8/site-packages/magic/__init__.py:422: PendingDeprecationWarning: Using compatability mode with libmagic's python binding. See https://github.com/ahupp/python-magic/blob/master/COMPAT.md for details.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED tests/exports/test_keepass.py::TestExportKeepassInsert::test_keepass_binary
FAILED tests/exports/test_keepass.py::TestExportKeepassInsert::test_keepass_insert
FAILED tests/exports/test_keepass.py::TestExportKeepassInsert::test_keepass_insert_empty
FAILED tests/imports/test_parse.py::TestParse::test_import_keepass - Assertio...
========== 4 failed, 110 passed, 1 deselected, 16 warnings in 17.47s ===========
builder for '/nix/store/da2g2vg3fa1vnw08dfdw9i62y1szvj5s-pass-import-3.1.drv' failed with exit code 1
```

**Additional context**

With `git bisect` I was able to narrow this down to the merge from `staging-next` that happened in 78af98a068906cb88fa14c31bd703341562bf890.

```
$ git bisect start --first-parent 1c65a509fbbc17a2853a657ea1391de0aab9e793 1fe6ed37fd9beb92afe90671c0c2a662a03463dd

$ git bisect run time nix-build -E 'with import <nixpkgs> { }; pkgs.callPackage ./pkgs/tools/security/pass/extensions/import.nix {}'
[...]
bisect run success

$ git bisect log
# bad: [1c65a509fbbc17a2853a657ea1391de0aab9e793] dico: 2.10 -> 2.11
# good: [1fe6ed37fd9beb92afe90671c0c2a662a03463dd] Merge pull request #120528 from marsam/update-goaccess
git bisect start '--first-parent' '1c65a509fbbc17a2853a657ea1391de0aab9e793' '1fe6ed37fd9beb92afe90671c0c2a662a03463dd'
# bad: [7bce67b46d6c3addd0c72ea20f83983efd4d1fa3] Merge pull request #120842 from peterhoeg/k/filemetadata
git bisect bad 7bce67b46d6c3addd0c72ea20f83983efd4d1fa3
# bad: [35dfa0e927924956cd47c083709a8147afea6d3d] alt-ergo: 2.3.3 → 2.4.0
git bisect bad 35dfa0e927924956cd47c083709a8147afea6d3d
# bad: [79cb2cb9869d7bb8a1fac800977d3864212fd97d] Merge pull request #120272 from fabaff/pynx584
git bisect bad 79cb2cb9869d7bb8a1fac800977d3864212fd97d
# good: [4a67076273c8c566f90e3f0caadce941106314d9] koka: make sure runtime dependencies are host -> target
git bisect good 4a67076273c8c566f90e3f0caadce941106314d9
# bad: [6e9671b6cb2341af2b3db9c8461aaf066239f7dc] Merge pull request #119960 from teto/openshift
git bisect bad 6e9671b6cb2341af2b3db9c8461aaf066239f7dc
# bad: [f24b888dc2efbd1f23bb6b1f46cee5e94c8fdd63] soldat-unstable: 2020-11-26 -> 2021-02-09
git bisect bad f24b888dc2efbd1f23bb6b1f46cee5e94c8fdd63
# good: [6eb1fc6f10143e177e24f09d71d47757fc874d2b] Merge pull request #120511 from LeSuisse/zookeeper-3.7.0
git bisect good 6eb1fc6f10143e177e24f09d71d47757fc874d2b
# bad: [78af98a068906cb88fa14c31bd703341562bf890] Merge pull request #119398 from NixOS/staging-next
git bisect bad 78af98a068906cb88fa14c31bd703341562bf890
# good: [7217b9e3c4e00f374b3d6b8b93390aa6ea177d39] Merge pull request #120503 from LeSuisse/exif-0.6.22
git bisect good 7217b9e3c4e00f374b3d6b8b93390aa6ea177d39
# first bad commit: [78af98a068906cb88fa14c31bd703341562bf890] Merge pull request #119398 from NixOS/staging-next
```


**Notify maintainers**
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

Hello pass-import maintainers! @lovek323 @fpletz @tadfisher

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.30, NixOS, 21.05.git.1fe6ed37fd9 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-20.03pre208171.2e8fc97dbfa"`
 - nixpkgs: `/etc/nixos/aij/unstable`


Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/python-modules/pykeepass/default.nix
+++ b/pkgs/development/python-modules/pykeepass/default.nix
@@ -1,15 +1,18 @@
-{ lib, fetchPypi, buildPythonPackage
+{ lib, fetchFromGitHub, buildPythonPackage
 , lxml, pycryptodomex, construct
 , argon2_cffi, dateutil, future
+, python
 }:
 
 buildPythonPackage rec {
   pname   = "pykeepass";
   version = "4.0.0";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "1b41b3277ea4e044556e1c5a21866ea4dfd36e69a4c0f14272488f098063178f";
+  src = fetchFromGitHub {
+    owner = "libkeepass";
+    repo = "pykeepass";
+    rev = version;
+    sha256 = "1zw5hjk90zfxpgq2fz4h5qzw3kmvdnlfbd32gw57l034hmz2i08v";
   };
 
   postPatch = ''
@@ -21,13 +24,15 @@ buildPythonPackage rec {
     argon2_cffi dateutil future
   ];
 
-  # no tests in PyPI tarball
-  doCheck = false;
+  checkPhase = ''
+    ${python.interpreter} -m unittest tests.tests
+  '';
 
-  meta = {
-    homepage = "https://github.com/pschmitt/pykeepass";
+  meta = with lib; {
+    homepage = "https://github.com/libkeepass/pykeepass";
+    changelog = "https://github.com/libkeepass/pykeepass/blob/${version}/CHANGELOG.rst";
     description = "Python library to interact with keepass databases (supports KDBX3 and KDBX4)";
-    license = lib.licenses.gpl3;
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ dotlambda ];
   };
-
 }

--- a/pkgs/tools/security/pass/extensions/import.nix
+++ b/pkgs/tools/security/pass/extensions/import.nix
@@ -17,9 +17,14 @@ python3Packages.buildPythonApplication rec {
     sha256 = "sha256-nH2xAqWfMT+Brv3z9Aw6nbvYqArEZjpM28rKsRPihqA=";
   };
 
-  # by default, tries to install scripts/pimport, which is a bash wrapper around "python -m pass_import ..."
-  # This is a better way to do the same, and takes advantage of the existing Nix python environments
   patches = [
+    (fetchpatch {
+      name = "support-for-keepass-4.0.0.patch";
+      url = "https://github.com/roddhjav/pass-import/commit/86cfb1bb13a271fefe1e70f24be18e15a83a04d8.patch";
+      sha256 = "0mrlblqlmwl9gqs2id4rl4sivrcclsv6zyc6vjqi78kkqmnwzhxh";
+    })
+    # by default, tries to install scripts/pimport, which is a bash wrapper around "python -m pass_import ..."
+    # This is a better way to do the same, and takes advantage of the existing Nix python environments
     # from https://github.com/roddhjav/pass-import/pull/138
     (fetchpatch {
       name = "pass-import-pr-138-pimport-entrypoint.patch";

